zap/src/test/java/org/parosproxy/paros/network/HttpRequestHeaderUnitTest.java

```diff
@@ -204,42 +209,117 @@ void shouldReplaceAnyCookieHeaderWhenSettingCookieParams() {
-                "http://example.org/css/file.ext", // In directory path
-                "http://ericsson.com/", // In domain name
-                "https://example.css", // In domain extension (TLD)
-                "https://example.css/dir/file.ext", // In domain extension (TLD)
-                "https://example.org/dir/file?foo=bar&thing=css", // In parameter value
-                "http://example.org/css/file.ext?foo=bar&type=.css", // In parameter value including
-                // period
-                "http://example.org/css/file.ext?foo=bar&thing=styles.css", // In parameter value,
-                // plausible filename
-                "https://example.org/dir/file?foo=bar&css=file.ext" // In parameter name
-            })
+    @MethodSource("falseTestCssUrls")
     void isCssShouldReturnFalseWhenUrlDoesNotIndicateCss(String url) {
         // Given
         HttpRequestHeader reqHeader = createRequestHeader(url);
         // When / Then
         assertFalse(reqHeader.isCss());
     }
 
+    private static Stream<Arguments> trueTestCssUrls() {
+        String[] exts = {"css"};
+        return trueTestUrls(exts);
+    }
+
     @ParameterizedTest
-    @ValueSource(
-            strings = {
-                "http://example.org/styles.css", // In path
-                "http://example.org/assets/css/styles.css", // In deeper path
-                "http://example.org/css/styles.css?foo=bar", // In path, ignoring params
-                "http://example.org/css/styles.css?foo=bar&thing=.css", // In path, ignoring params
-            })
+    @MethodSource("trueTestCssUrls")
     void isCssShouldReturnTrueWhenUrlIndicatesCss(String url) {
         // Given
         HttpRequestHeader reqHeader = createRequestHeader(url);
         // When / Then
         assertTrue(reqHeader.isCss());
     }
 
+    private static Stream<Arguments> falseTestImageUrls() {
+        return falseTestUrls("gif");
+    }
+
+    @ParameterizedTest
+    @MethodSource("falseTestImageUrls")
+    void isImageShouldReturnFalseWhenUrlDoesNotIndicateImage(String url) {
+        // Given
+        HttpRequestHeader reqHeader = createRequestHeader(url);
+        // When / Then
+        assertFalse(reqHeader.isImage());
+    }
+
+    private static Stream<Arguments> trueTestImageUrls() {
+        // Per HttpRequestHeader.patternImage
+        String[] exts = {"bmp", "ico", "jpg", "jpeg", "gif", "tiff", "tif", "png", "svg"};
+        return trueTestUrls(exts);
+    }
+
+    @ParameterizedTest
+    @MethodSource("trueTestImageUrls")
+    void isImageShouldReturnTrueWhenUrlIndicatesImage(String url) {
+        // Given
+        HttpRequestHeader reqHeader = createRequestHeader(url);
+        // When / Then
+        assertTrue(reqHeader.isImage());
+    }
+
+    private static Stream<Arguments> falseTestUrls(String extension) {
+        List<Arguments> urls = new ArrayList<Arguments>();
+        urls.add(arguments("http://example.org/" + extension + "/file.ext")); // In directory path
```

Comment:
This is from existing code but it's better if the comments are added in the previous line to avoid the formatting breaks.